install-panel.sh

```diff
@@ -711,49 +682,131 @@ function main {
+  echo -e -n "\n* Initial configuration completed. Continue with installation? (y/N): "
   read -r CONFIRM
   if [ "$CONFIRM" == "y" ]; then
     perform_install
+  elif [ "$CONFIRM" == "Y" ]; then
+    exit 0
   elif [ "$CONFIRM" == "n" ]; then
     exit 0
+  elif [ "$CONFIRM" == "N" ]; then
+    exit 0
   else
     # run welcome script again
-    print_error "Invalid confirm. Will exit."
+    print_error "Invalid confirm. Will repeat."
     exit 1
   fi
+}
+
+function ufw-Ubuntu {
+  #This function is to use UFW on Ubuntu to Open ports based on whether the user is going
+  #to be hosting the daemon on their machine or just the panel.
+  #When you want to call the total UFW function, you call this function
+
+  read -p "Are you going to be installing the Daemon on this machine (y/N): " DaemonAsk
+
+  #If you are going to be installing the daemon and you want a firewall, by default the 
+  #ports 80, 22, 2022 and 8080 Are allowed through the firewall
+  if [[ "$DaemonAsk" =~ [Yy] ]]; then
+    read -p "Do you want to enable a firewall? (y/N): " ConfirmFirewall
+    if [[ "$ConfirmFirewall" =~ [Yy] ]]; then
+      apt install ufw -y
+      echo -e "\nEnabling Uncomplicated Firewall (UFW)"
+      echo " This is to do the inital setup of UFW, opening the following ports."
+      echo " If you're running SSL you need to add the port 443"
+      echo -e "  80\n  22\n  2022\n  8080"
+      read -p "Would you like to add additional ports besides the ones listed? (y/N): " ConfirmAdditional
+
+      #This is where it asks the user if they want to add any additional ports
+      if [[ "$ConfirmAdditional" =~ [Yy] ]]; then
+        ufwAdditional-Ubuntu
+        ufw allow 80 | grep 'zzz'
+        ufw allow 22 | grep 'zzz'
+        ufw allow 2022 | grep 'zzz'
+        ufw allow 8080 | grep 'zzz'
+        ufw enable
+        ufw status numbered | sed '/v6/d'
+        sleep 3
+      fi
+    fi
+  else
+
+    #If you aren't going to be installing the daemon and you want a firewall, only the
+    #ports 22 and 80 are opened through ufw.
+    read -p "Do you want to enable a firewall? (y/N): " ConfirmFirewall
+    if [[ "$ConfirmFirewall" =~ [Yy] ]]; then
+      apt install ufw -y
+      echo -e "\nEnabling Uncomplicated Firewall (UFW)"
+      echo " This is to do the inital setup of UFW, opening the following ports."
+      echo " If you're running SSL you need to add the port 443"
+      echo -e "  80\n  22"
+      read -p "Would you like to add additional ports besides the ones listed? (y/N): " ConfirmAdditional
+
+      #This is where it asks the user if they want to add any additional ports
+      if [[ "$ConfirmAdditional" =~ [Yy] ]]; then
+        ufwAdditional-Ubuntu
+        ufw allow 80 | grep 'zzz'
+        ufw allow 22 | grep 'zzz'
+        ufw enable
+        ufw status numbered | sed '/v6/d'
+        sleep 3
+      fi  
+    fi
+  fi
+}
 
+function ufwAdditional-Ubuntu {
```

Comment:
This is not necessary to the script. Ports related to Pterodactyl should only be opened. I reckon this function should be removed.